src/main/java/org/fungover/thunder/Client.java

```diff
@@ -0,0 +1,33 @@
+package org.fungover.thunder;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Client {
+    private final String clientId;
+    private boolean connected;
+    private Set<String> subscribedTopics;
+
+    public Client(String clientId) {
+        this.clientId = clientId;
+        this.connected = false;
+        this.subscribedTopics = new HashSet<>();
+    }
```

Comment:
```suggestion
public class Client {
    private final String clientId;
    private boolean connected;
    private Set<String> subscribedTopics;
    private Socket socket;

    public Client(String clientId) {
        this.clientId = clientId;
        this.connected = false;
        this.subscribedTopics = new HashSet<>();
        this.socket = new Socket();
    }
```
One thing I am missing is the reference to a socket so we can read from and send to the client. Other than that I think it looks good.